behave_framework/src/minifi_test_framework/containers/container.py

```diff
@@ -0,0 +1,266 @@
+#
+#  Licensed to the Apache Software Foundation (ASF) under one or more
+#  contributor license agreements.  See the NOTICE file distributed with
+#  this work for additional information regarding copyright ownership.
+#  The ASF licenses this file to You under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with
+#  the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+import logging
+import os
+import shlex
+import tempfile
+from docker.models.networks import Network
+from minifi_test_framework.containers.directory import Directory
+from minifi_test_framework.containers.file import File
+from minifi_test_framework.containers.host_file import HostFile
+from typing import Dict, Any, List
+
+import docker
+
+
+class Container:
+    def __init__(self, image_name: str, container_name: str, network: Network):
+        self.image_name: str = image_name
+        self.container_name: str = container_name
+        self.network: Network = network
+        self.user: str = "0:0"
+        self.client = docker.from_env()
+        self.container = None
+        self.files: List[File] = []
+        self.dirs: List[Directory] = []
+        self.host_files: List[HostFile] = []
+        self.volumes = {}
+        self.command = None
+        self._temp_dir = None
+        self.ports = None
+        self.environment: List[str] = []
+
+    def deploy(self) -> bool:
+        self._temp_dir = tempfile.TemporaryDirectory()
+
+        if len(self.files) != 0 or len(self.dirs) != 0 or len(self.host_files) != 0:
```

Comment:
Do we need this check if we just iterate through all of the files? If those lists are empty we don't do anything anyway